instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollector.java

```diff
@@ -66,43 +73,52 @@ static void registerObservers(
       Function<Notification, GarbageCollectionNotificationInfo> notificationInfoExtractor) {
     Meter meter = RuntimeMetricsUtil.getMeter(openTelemetry);
 
-    LongHistogram gcTime =
+    DoubleHistogramBuilder gcDurationBuilder =
         meter
             .histogramBuilder("process.runtime.jvm.gc.duration")
             .setDescription("Duration of JVM garbage collection actions")
-            .setUnit("ms")
-            .ofLongs()
-            .build();
+            .setUnit("s");
+    setGcDurationBuckets(gcDurationBuilder);
+    DoubleHistogram gcDuration = gcDurationBuilder.build();
 
     for (GarbageCollectorMXBean gcBean : gcBeans) {
       if (!(gcBean instanceof NotificationEmitter)) {
         continue;
       }
       NotificationEmitter notificationEmitter = (NotificationEmitter) gcBean;
       notificationEmitter.addNotificationListener(
-          new GcNotificationListener(gcTime, notificationInfoExtractor), GC_FILTER, null);
+          new GcNotificationListener(gcDuration, notificationInfoExtractor), GC_FILTER, null);
+    }
+  }
+
+  private static void setGcDurationBuckets(DoubleHistogramBuilder builder) {
+    if (!(builder instanceof ExtendedDoubleHistogramBuilder)) {
+      // that shouldn't really happen
```

Comment:
If that's true than you shouldn't have to guard against it, eh? 😉

Reply:
Better safe than sorry 💯